docs/src/adr/025adr-transpile-target.md

```diff
@@ -0,0 +1,102 @@
+- [Code changes](#code)
+- [Testing](#testing)
+- [Summary](#summary)
+
+## Overview
+CHC is to be an alternate transpilation target to the already available vmt option in Apalache.
+Given this, a new option is to be added to the `transpile` command of the CLI in addition to adding support for integers .
+
+## CLI Transpilation Target Option
+
+Added `--transpile-target` flag for chc (no envvar equivalent yet), whose default value will be `vmt`.
+The option description is shown below.
+
+```
+--transpile-target : the transpilation target: vmt, chc (experimental), default: vmt
+```
+
+### Code changes
+
+The following changes will be made to add target formalism option to the transpile command,
+add support for chc transpile, arithmetics and invariants in chc transpilation 
+and implement this in the CLI option:
+
+#### Target formalism option to the transpile command and CLI option implementation
+
+- Add new class `Transpiler` with `@param` `transpilationTarget` in `options.scala` 
+  to specify the transpilation target options added under the new class `TranspilationTarget` 
+  with `TranspilationTarget.VMT` and `TranspilationTarget.CHC` for vmt and chc respectively.
+- Add new string `var` `transpilationTarget` to class `TranspileCmd` to add support for chc encoding in the CLI using the `--transpile-target` flag.
+
+#### Support for CHC transpile
+
+- Add new class `ReTLACombinedPredicateForCHC` extending `ReTLACombinedPredicate` to use in `ReTLAToVMTModule` with
+  new `ReTLALanguagePredForCHC` to test whether the expression fits into the reTLA fragement.
+- Add `ReTLALanguagePredForCHC` extending `ReTLALanguagePred` to include `TlaBoolOper` and `TlaArithOper`.
+- Add new class `ReTLAPreproPassImplForCHC` that extends `ReTLAPreproPassImpl` 
+  to simplify TLA+ expressions to chc by running multiple transformations supported with `ReTLALanguagePredForCHC`.
+- Add new class `ReTLAToCHCModule` extending `ReTLAToVMTModule` to bind the chc classes together and
+  transpiles reTLA input to chc.
+- Add new class `ReTLAToCHCTranspilePassImpl` with `TlaExToCHCWriter` to handle the last step of transpiling
+  by assembling the .chc output file. Given a disassembled module (extracted init, next, inv, etc.), 
+  CHCWriter determines which parts get which `VMTExpr` wrappers. Then, using TermToCHCWriter, 
+  it translates the terms to output strings, and adds sort/function declarations and definitions.
+- Add `TermToCHCWriter` to manage the translation of terms to strings to be written to the final output file.
+- Add new class `ToTermRewriterImplForCHC` extending `ToTermRewriterImpl` to implement the `ToTermRewriter` from reTLA to SMT Terms with additional
+  support for arithmetics with a new `IntRuleForCHC` and `BoolRuleForCHC`.
+- Add new class `ValueRuleForCHC` extending `ValueRule` to handle the conversion of literals to names. 
+  It renames all variables from  `x'` to `x.prime` for chc.
+
+#### Support for arithmetics and invariants
```

Comment:
what about the --temporal flag?

Reply:
I have tested the --temporal flag. No issues with that too!